### PR TITLE
Bug 1846287 - Catch LoginsApiException exception when fetching the list of logins and setting startup metrics

### PR DESCRIPTION
**Pull request by gabrielluong**
STRs:
1. Call FenixApplication$setStartupMetrics
2. Call `Logins.savedAll.set(lazyPasswordStorage.value.list().size.toLong())` https://searchfox.org/mozilla-mobile/rev/c10f8320ab3f881713245c7f3434b21d021e8abf/firefox-android/fenix/app/src/main/java/org/mozilla/fenix/FenixApplication.kt#843-844
3. We get an incorrect key error from AS when calling `list()` and the subsequent `decryptLogin()` call here  https://searchfox.org/mozilla-mobile/rev/c10f8320ab3f881713245c7f3434b21d021e8abf/firefox-android/android-components/components/service/sync-logins/src/main/java/mozilla/components/service/sync/logins/SyncableLoginsStorage.kt#177. This call can throw an exception so we should handle it and fail gracefully.

Solution is to add a try catch around https://searchfox.org/mozilla-mobile/rev/c10f8320ab3f881713245c7f3434b21d021e8abf/firefox-android/fenix/app/src/main/java/org/mozilla/fenix/FenixApplication.kt#843-844

Initially, I was wondering if we should (1) catch the `LoginsApiException` and log the exception and do nothing with the telemetry or (2) catch the exception and still report telemetry but with an empty list for the logins.

After talking with @HarrisonOg, it made more sense to go with option 1 - catch and log the exception and not send telemetry. Having the `logins.savedAll` not be sent will better indicate an issue with login telemetry than sending an empty set which could be interpreted as the user having no saved logins.

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1846287

### CODE REVIEW

File: fenix/app/src/main/java/org/mozilla/fenix/FenixApplication.kt [hunk 1]
```diff
@@ -463,7 +464,7 @@ open class FenixApplication : LocaleAwareApplication(), Provider {
         // We need the push feature setup here to deliver messages in the case where the service
         // starts up the app first.
         components.push.feature?.let {
-            Logger.info("AutoPushFeature is configured, initializing it...")
+            logger.info("AutoPushFeature is configured, initializing it...")
```
Comment (gabrielluong) [inline fenix/app/src/main/java/org/mozilla/fenix/FenixApplication.kt:467]: Cleaning up some of these individual `Logger` to prefer our tagged Logger defined in https://searchfox.org/mozilla-mobile/source/firefox-android/fenix/app/src/main/java/org/mozilla/fenix/FenixApplication.kt#137

**Review by HarrisonOg — APPROVED** [review]

**Comment by HarrisonOg** [comment]
looks good to me! 🚢

**Comment by gabrielluong** [comment]
@Mergifyio backport releases_v116

**Comment by gabrielluong** [comment]
@Mergifyio backport releases_v117

**Pull request merged**